chroma-manager/chroma_core/services/job_scheduler/command_plan.py

```diff
@@ -346,6 +346,12 @@ def _emit_transition_deps(self, transition, transition_stack = {}):
         transition_stack[transition.stateful_object] = transition.new_state
         log.debug("Updating transition_stack[%s/%s] = %s" % (transition.stateful_object.__class__, transition.stateful_object.id, transition.new_state))
 
+        # do nothing for a NOOP transition
+        if transition.old_state == transition.new_state:
```

Comment:
This seems like a gap we would have seen many times. Are you sure there's not some other edge case being missed?

Reply:
That is **exactly** what I was thinking and I am far from an expert in this area of the code, so the more eyes the better.

But this simple fix does make [test_mount_and_unmount](https://github.com/intel-hpdd/intel-manager-for-lustre/blob/92f459306fdf0bb26528e8a55a236ab0e35531e2/chroma-manager/tests/integration/shared_storage_configuration/test_client_mount_management.py#L22) pass where it was previously failing with [HYD-3468](https://jira.hpdd.intel.com/browse/HYD-3468).

I've started up 5x SSI testing with this patch underneath #250 so we'll see how it fares.

Reply:
It had one actual honest pass and the other 4 failures were all for other known intermittent issues.